.github/workflows/sonar.yml

```diff
@@ -0,0 +1,201 @@
+name: Run sonar analysis
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  SONAR_URL: https://sonarcloud.io
+  SONAR_ORGANIZATION: telecominfraproject
+  MAVEN_OPTS: -Xmx3g -Xss2m # prevents java.lang.StackOverflowError
+
+jobs:
+  sonar:
+    strategy:
+      matrix:
+        project:
```

Comment:
Looking at this matrix now - we will add more projects to this repository. Having this matrix explicitly defined here creates a need tor manual sync whenever new projects are added. Can we somehow auto-generate the matrix? Or rely on maven dependencies to get all the projects?

Reply:
okay I added a workaround with dispatch events